### ISSUE TITLE
RTT mip-maps are not being generated automatically on D3D12

### ISSUE THREAD

**Issue by docEdub** [BabylonJS/BabylonNative#1265]
Running playground https://playground.babylonjs.com/#W7E7CF#25 on D3D12 shows the following when the window is small or the camera is moved far enough away...

![MicrosoftTeams-image (4)](https://github.com/BabylonJS/BabylonNative/assets/78323754/45f8dd50-9cf3-4c03-8213-348b7915293a)

It should look like this ...

![](https://github.com/BabylonJS/BabylonNative/assets/78323754/2d91c003-c108-418b-acdb-d784899c2eae)

Cedric and I looked at this in RenderDoc and saw the RTT mip-maps are missing on D3D12. They are generated automatically for other graphics APIs, but not D3D12.

**Comment by CedricGuillemet** [comment]
I believe we can close this issue ? @docEdub

**Comment by docEdub** [comment]
> I believe we can close this issue ? @docEdub

I don't think this is fixed, yet.

**Comment by docEdub** [comment]
I found this bgfx issue that may be related: https://github.com/bkaradzic/bgfx/issues/798

TLDR; Automatic mip-map generation is not provided by D3D12, Vulkan, or Metal, so it needs to be explicitly implemented in bgfx to match the behavior seen in other graphics APIs. For Vulkan, this was implemented in bgfx in PR https://github.com/bkaradzic/bgfx/pull/2472, but it has not been implemented in D3D12 or Vulkan, yet, and afaict there is no plan to do so.

The end-result is the mip-map layers get created by bgfx on D3D12, but they are all black.
